server/sonar-server/src/main/java/org/sonar/server/permission/ws/WsProjectRef.java

```diff
@@ -29,7 +29,7 @@
  * Project identifiers from a WS request. Guaranties the project id and project key are not provided at the same time.
  */
 public class WsProjectRef {
-  private static final String MSG_ID_OR_KEY_MUST_BE_PROVIDED = "Project id or project key must be provided, not both.";
+  private static final String MSG_ID_OR_KEY_MUST_BE_PROVIDED = "Project id or project key can be provided, not both.";
```

Comment:
This is why I don't like WS validation to rely on code in beans.

There actually is usage of this class where project reference is mandatory (ws api/permission/apply_template) and "must" is correct and you shouldn't have changed it. There is also other place where it is not mandatory and "can" is correct (WebServices relying on `PermissionWsSupper#findProject`).